fenix/app/src/main/java/org/mozilla/fenix/HomeActivity.kt

```diff
@@ -1175,6 +1185,54 @@ open class HomeActivity : LocaleAwareAppCompatActivity(), NavHostActivity {
             !processIntent(intent)
     }
 
+    private fun showFullscreenMessageSurveyIfNeeded(context: Context) {
+        val messagingStorage = context.components.analytics.messagingStorage
+        val messages = runBlockingIncrement { messagingStorage.getMessages() }
+        val nextMessage =
+            messagingStorage.getNextMessage(FenixMessageSurfaceId.SURVEY, messages)
+                ?: return
+
+        if (nextMessage.id == "fullscreen-viewpoint-invitation-message") {
```

Comment:
We do in this case.  Since `FenixMessageSurfaceId.SURVEY` can be used for either the homescreen survey or the fullscreen survey.  The id is the one to determine which one we're showing.

Reply:
I'll double check.  I was worried that we might use the homescreen survey but you're right this won't happen.